src/main/java/com/powsybl/openloadflow/network/action/LfHvdcAction.java

```
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2025, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.network.action;
+
+import com.powsybl.action.HvdcAction;
+import com.powsybl.openloadflow.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+/**
+ * @author Bertrand Rix {@literal <bertrand.rix at artelys.com>}
+ * @author Anne Tilloy {@literal <anne.tilloy at rte-france.com>}
+ * @author Jean-Luc Bouchot {@literal <jlbouchot at gmail.com>}
+ */
+public class LfHvdcAction extends AbstractLfAction<HvdcAction> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LfHvdcAction.class);
+
+    public LfHvdcAction(String id, HvdcAction action) {
+        super(id, action);
+    }
+
+    @Override
+    public boolean apply(LfNetwork network, LfContingency contingency, LfNetworkParameters networkParameters) {
+        LfHvdc lfHvdc = network.getHvdcById(action.getHvdcId());
```

I'm adding here a question about the architecture change (not only specific on HVDC).
This change is not purely about architecture.
It changes also at what time static verifications (independant from the contingence) are done.
Before:
   - Where done at action creation time (once per SA run). In case of issue the action was silently ignored (now they are reproted and it is better)
With this PR:
   - Verifications are performed for each contingence that needs them (innefficient - or can the action validity depend on the contingence ?)
   - But the result in case of inconsitency is displayed in context of the contingence via a report (good)

Can't we get the best of both world by checking the action validity at creation time, and generating the report at apply time ?

---

Good questions !

I can think of some scenario where the action application validity could depend on the contingency :

* With the work currently being done on load flow parameters specific to some contingencies : might impact the succesful application of an action.
* With some work planned on Artelys side to take into account the equivalent of "auto" action / SPS from a RAO that might be dependent on a type of contingency and then impact differently a same action done in different N-1 scenario. Maybe that's a bit early to care about that though but thats a possibility. 
* More simply an action trying to impact an equipment that was indirectly impacted by a specific contingency

I think most checks / actions application are very fast (a lookup + eventually a boolean check + a call to one or more setter). I am not too worried about that at the moment.
Some of the checks like null branch, null area could probably be removed from the apply method because they are redondant with the checkActions function though.

Maye be some checks can definitely be moved at construction time like you suggested for the generator. I'll look at it again.